object-store-client/src/main/java/collaboratory/storage/object/store/client/exception/ServiceRetryableResponseErrorHandler.java

```diff
@@ -37,18 +37,22 @@ public void handleError(ClientHttpResponse response) throws IOException {
     switch (response.getStatusCode()) {
     case NOT_FOUND:
     case BAD_REQUEST:
-      log.warn("Not Retryable Endpoint: {}", response.getStatusText());
-      throw new NotRetryableException(new IOException("object store service error: "
+      log.warn("Bad request. Stop processing: {}", response.getStatusText());
+      throw new NotRetryableException(new IOException("Object store client error: "
           + IOUtils.toString(response.getBody())));
     case INTERNAL_SERVER_ERROR:
       log.warn("Server error. Stop processing: {}", response.getStatusText());
-      throw new NotResumableException(new IOException("Object store service error: "
+      throw new NotResumableException(new IOException("Object store client error: "
+          + IOUtils.toString(response.getBody())));
+    case UNAUTHORIZED:
```

Comment:
This was missing. Really we should not be retrying any `4XX` series error codes as well.